zipline/pipeline/loaders/quarter_estimates.py

```diff
@@ -0,0 +1,582 @@
+                append=True
+            ).index
+            requested_qtr_data = self.get_requested_quarter_data(
+                stacked_last_per_qtr, requested_qtr_idx, dates
+            )
+
+            # Calculate all adjustments for the given quarter and accumulate
+            # them for each column.
+            col_to_adjustments = self.get_adjustments(zero_qtr_data,
+                                                      requested_qtr_data,
+                                                      last_per_qtr,
+                                                      dates,
+                                                      assets_with_data,
+                                                      columns)
+            for col in columns:
+                column_name = self.name_map[col.name]
+                # We may have dropped assets if they never have any data for
+                # the requested quarter.
+                df = pd.DataFrame(data=requested_qtr_data[column_name],
+                                  index=dates,
+                                  columns=assets,
+                                  dtype=col.dtype)
+
+                out[col] = AdjustedArray(
+                    df.values.astype(col.dtype),
+                    mask,
+                    dict(col_to_adjustments[column_name]),
+                    col.missing_value,
+                )
+        return out
+
+    def get_last_data_per_qtr(self, assets_with_data, columns, dates):
+        """
+        Determine the last piece of information we know for each column on each
+        date in the index for each sid and quarter.
+
+        Parameters
+        ----------
+        assets_with_data : pd.Index
+            Index of all assets that appear in the raw data given to the
+            loader.
+        columns : iterable of BoundColumn
+            The columns that need to be loaded from the raw data.
+        dates : pd.DatetimeIndex
+            The calendar of dates for which data should be loaded.
+
+        Returns
+        -------
+        stacked_last_per_qtr : pd.DataFrame
+            A DataFrame indexed by [dates, sid, normalized_quarters] that has
+            the latest information for each row of the index, sorted by event
+            date.
+        last_per_qtr : pd.DataFrame
+            A DataFrame with columns that are a MultiIndex of [
+            self.estimates.columns, normalized_quarters, sid].
+        """
+        # Get a DataFrame indexed by date with a MultiIndex of columns of [
+        # self.estimates.columns, normalized_quarters, sid], where each cell
+        # contains the latest data for that day.
+        last_per_qtr = last_in_date_group(
+            self.estimates, dates, assets_with_data, reindex=True,
+            extra_groupers=[NORMALIZED_QUARTERS]
+        )
+        # Forward fill values for each quarter/sid/dataset column.
+        ffill_across_cols(last_per_qtr, columns, self.name_map)
```

Comment:
Fixed this in `ffill_across_cols` by calling `unstack` in a loop. The problem was that calling `unstack` once with a list of levels as an argument was creating a DF with object columns due to a bug in pandas.